sdk/eventhub/eventhubs-checkpointstore-blob/README.md

```diff
@@ -1,19 +1,105 @@
 
 ## Getting started
 
+### Install the package
+
+Install the Azure Event Hubs Checkpoint Store Blob library using npm
+
+`npm install @azure/eventhubs-checkpointstore-blob@latest`
+
+**Prerequisites**: You must have an [Azure subscription](https://azure.microsoft.com/free/) and a
+[Event Hubs Namespace](https://docs.microsoft.com/en-us/azure/event-hubs/) to use this package, and a [Storage account](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction)
+If you are using this package in a Node.js application, then use Node.js 8.x or higher.
+
+### Configure Typescript
+
+TypeScript users need to have Node type definitions installed:
+
+```bash
+npm install @types/node
+```
+
+You also need to enable `compilerOptions.allowSyntheticDefaultImports` in your tsconfig.json. Note that if you have enabled `compilerOptions.esModuleInterop`, `allowSyntheticDefaultImports` is enabled by default. See [TypeScript's compiler options handbook](https://www.typescriptlang.org/docs/handbook/compiler-options.html) for more information.
+
 ## Key concepts
 
 ## Examples
 
+### Persistent event checkpointing with Azure Blob storage
+
+To consume events from all the partitions of an Event Hub, you'll create an [EventProcessor](https://azure.github.io/azure-sdk-for-js/event-hubs/classes/eventprocessor.html)
+for a specific consumer group. When an Event Hub is created, it provides a default consumer group that can be 
+used to get started.
+
+The `EventProcessor` will delegate the processing of events to a [PartitionProcessor](https://azure.github.io/azure-sdk-for-js/event-hubs/classes/partitionprocessor.html)
+that you provide, allowing you to focus on business logic while the processor holds responsibility for managing the underlying consumer
+operations including checkpointing and load balancing.
+
+If you want to store checkpoints in [Azure Storage Blobs](https://azure.microsoft.com/en-us/services/storage/blobs/), you can provide an instance of [BlobPartitionManager](https://azure.github.io/azure-sdk-for-js/eventhubs-checkpointstore-blob/classes/blobpartitionmanager.html) to your Event Processor. `BlobPartitionManager` uses Storage Blobs to store checkpoints and balance partition load among all instances of Event Processors.
+
+you can see how to use the `EventProcessor` in the below
+example, where we use `BlobPartitionManager` that does 
+checkpointing in [Azure Storage Blobs](https://azure.microsoft.com/en-us/services/storage/blobs/).
+
+```javascript
+const client = new EventHubClient("my-connection-string", "my-event-hub");
+const containerClient = new ContainerClient("storage-connection-string", "container-name");
+await containerClient.create();
+
+const processor = new EventProcessor(
+  EventHubClient.defaultConsumerGroupName,
+  client,
+PartitionProcessor,
```

Comment:
@chradek, I know we discussed using the PartitionProcessor that will just log, but this will only showcase instantiating classes and not the power of the partition manager.

I want to suggest having 2 examples:
- Create a Partition Manager using Azure Blob Storage
- Checkpoint events using Azure Blob storage

Reply:
### Create a Partition Manager using Azure Blob Storage

To create a Partition Manager that uses Azure Blob Storage, you need to first have an [Azure subscription](https://azure.microsoft.com/free/) and an [Azure Storage account](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction)

Then, you can use the below code snippet to create a Partition Manager

```javascript
import { ContainerClient } from "@azure/storage-blob",
import { BlobPartitionManager } from "@azure/eventhubs-checkpointstore-blob"

const containerClient = new ContainerClient("storage-connection-string", "container-name");
await containerClient.create(); // This can be skipped if the container already exists

const partitionManager =  new BlobPartitionManager(containerClient);
```

### Checkpoint events using Azure Blob storage

To checkpoint events received using an EventProcessor to Azure Blob Storage, you will need to pass an instance of the Partition Manager to the Event Processor along with the code to call the `updateCheckpoint()` method.

In this example, we will use a `SamplePartitionProcessor` that extends the `PartitionProcessor` class in order to checkpoint the last event in the batch.

```javascript
import { ContainerClient } from "@azure/storage-blob",
import { BlobPartitionManager } from "@azure/eventhubs-checkpointstore-blob"
import { EventHubClient, PartitionProcessor } from "@azure/event-hubs"

const eventhubClient = new EventHubClient("event-hub-connectionstring")
const containerClient = new ContainerClient("storage-connection-string", "container-name");
await containerClient.create(); // This can be skipped if the container already exists
const partitionManager =  new BlobPartitionManager(containerClient);

SamplePartitionProcessor extends PartitionProcessor  {
   // add code here to override the processevents method to checkpoint the last event in the batch
}

const eventProcessor = new EventProcessor(....)
eventProcessor.start()
```

Reply:
In `Prerequisites` section we already talked about - "you need to first have an [Azure subscription](https://azure.microsoft.com/free/) and an [Azure Storage account](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction)"

Do we need this here as well?

Reply:
No, we can skip that part